### PR TITLE
Add "Rediscovering the Browser" session

### PR DESCRIPTION
**Pull request by teodutu**

### CODE REVIEW

File: rediscovering-the-browser/index.md [hunk 1]
```diff
@@ -12,29 +12,559 @@ Use [gh-md-toc](https://github.com/ekalinin/github-markdown-toc).
...
+#### Authorisation
+
+In order to have a finer access control, such as specifying that a given user may read some files from the server while others can't, authentication is not enough.
+We need a means to **enforce** some security policies like, for example, UNIX-like permissions on the file system.
+Authentication helps us identify the user, but we need another "A", called **Authorisation** in order to decide what that user may or may not do.
+
+In short, authorisation answers the question: "What can you do?".
+It means defining a set of **policies** by which to grant users various kinds of access to different resources.
+You can read about the 3 large types of access control types, from which policies derive, [at the end of this session](#access-control-types). 
+
```
Comment (gabrielmocanu) [inline rediscovering-the-browser/index.md:75]: Where is accounting?

**Review by gabrielmocanu — CHANGES_REQUESTED** [review]

File: rediscovering-the-browser/index.md [hunk 2]
```diff
@@ -12,29 +12,559 @@ Use [gh-md-toc](https://github.com/ekalinin/github-markdown-toc).
...
+So the request would look something like:
+```
+GET /path/to/myfile.html HTTP/1.1
+[...]
+```
+
+Below the first line in the picture that precedes the anatomy of a URL, we can find the **headers** of the request.
+They are metadata used to provide additional information about the connection, about the client and about how to handle the request.
+Some usual headers are:
+- **Host:** indicates the desired host handling the request
```
Comment (gabrielmocanu) [inline rediscovering-the-browser/index.md:179]: Add content type and content length.

File: rediscovering-the-browser/index.md [hunk 3]
```diff
@@ -12,29 +12,559 @@ Use [gh-md-toc](https://github.com/ekalinin/github-markdown-toc).
...
+- `1xx`: informational responses
+- `2xx`: the request was fulfilled successfully.
+The most common code is `200`.
+- `3xx`: redirects - the request was passed to another server
+- `4xx`: client errors.
+Some very common client errors are:
+    - `400`: **bad request** - there's an error in the request.
+    - `404`: **not found** - the requested resource doesn't exist.
+    For example, in case `/file.txt` doesn't exist and the client sends `GET /file.txt HTTP/1.1`, the server answers with `404`.
+    - `403`: **unauthorised** - you don't have access to that resource.
```
Comment (gabrielmocanu) [inline rediscovering-the-browser/index.md:201]: Order them, it would looks better.

**Review by gabrielmocanu — CHANGES_REQUESTED** [review]

File: rediscovering-the-browser/index.md [hunk 4]
```diff
@@ -12,29 +12,559 @@ Use [gh-md-toc](https://github.com/ekalinin/github-markdown-toc).
...
+It's something similar to escaping characters in bash.
+
+Specifically, in the URL below:
+- `+` is the URL encoding for space
+- `%3B` is the URL encoding for `;`
+- `%3F` is the URL encoding for `?`
+
+### Tutorial - Readme
+
+The website tells you ti click a hyperlink.
```
Comment (gabrielmocanu) [inline rediscovering-the-browser/index.md:372]:
```suggestion
The website tells you to click a hyperlink.
```

File: rediscovering-the-browser/index.md [hunk 5]
```diff
@@ -12,29 +12,559 @@ Use [gh-md-toc](https://github.com/ekalinin/github-markdown-toc).
...
+Do it and then look at the address bar.
+Notice the query parameter `flag=SSS_CTF{do_not_kill_the_messenger}`.
```
Comment (gabrielmocanu) [inline rediscovering-the-browser/index.md:374]: Should the flag be here?

**Review by gabrielmocanu — CHANGES_REQUESTED** [review]

File: rediscovering-the-browser/index.md (hunk 3 above)
Comment (teodutu) [reply rediscovering-the-browser/index.md:201]: Do you mean to write them in the order 400, 403, 404, 405 or to use a numbered list? I chose this ordering to explain 403 based on 404.

Comment (gabrielmocanu) [reply rediscovering-the-browser/index.md:201]: Oh. Okey then.

File: rediscovering-the-browser/index.md (hunk 5 above)
Comment (teodutu) [reply rediscovering-the-browser/index.md]: No, I've already removed it.

File: rediscovering-the-browser/index.md (hunk 1 above)
Comment (teodutu) [reply rediscovering-the-browser/index.md:75]: FML. I've added it now. Thanks!

**Review by gabrielmocanu — APPROVED** [review]

**Pull request merged**